drivers/amazonec2/amazonec2.go

```diff
@@ -310,17 +310,18 @@ func (d *Driver) Create() error {
 
 	d.InstanceId = instance.InstanceId
 	log.Debug("waiting for ip address to become available")
-	for {
+	if err := utils.WaitFor(func() bool {
```

Comment:
I think pulling the anonymous function out of the if statement would make this code much more readable.